File: 摘要算法/hongren点集/hongren.js
n = {
    "no": "dy3009",
    "data": {}
}
const cryptoJS = require("crypto-js")

get_params = function (text) {
    var e = (new Date).getTime()
    C = "kbn%&)@<?FGkfs8sdf4Vg1*+;`kf5ndl$"
    // text = "param=" + JSON.stringify(text) + "&timestamp=" + e + "&tenant=1&salt=" + C
    //param="{\"no\":\"dy3009\",\"data\":{}}"&timestamp=1716996279382&tenant=1&salt=kbn%&)@<?FGkfs8sdf4Vg1*+;`kf5ndl$  XXXXX错误XXXXX
    text = "param=" + text + "&timestamp=" + e + "&tenant=1&salt=" + C
    console.log(text);
    sign = cryptoJS.SHA256(text).toString()
    return [e, sign]
    // console.log(cryptoJS.SHA256("text").toString());
}

console.log(get_params("{\"no\":\"dy3009\",\"data\":{}}"));


// console.log(cryptoJS.SHA256("1").toString());

File: 摘要算法/hongren点集/hongren.py
import execjs
import requests
import json
headers = {
    'Accept': 'application/json, text/plain, */*',
    'Accept-Language': 'zh-CN,zh;q=0.9',
    'Cache-Control': 'no-cache',
    'Connection': 'keep-alive',
    'Content-Type': 'application/json;charset=UTF-8',
    'Origin': 'https://www.hh1024.com',
    'Pragma': 'no-cache',
    'Sec-Fetch-Dest': 'empty',
    'Sec-Fetch-Mode': 'cors',
    'Sec-Fetch-Site': 'cross-site',
    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36',
    'sec-ch-ua': '"Not)A;Brand";v="99", "Google Chrome";v="127", "Chromium";v="127"',
    'sec-ch-ua-mobile': '?0',
    'sec-ch-ua-platform': '"Windows"',
}

json_data = {
    'param': '{"no":"dy3009","data":{}}',
    # 'sign': '9d19abacffe0b16f5c0569ac588cbab58418a8f82d034e92dcab2865cea38420',
    # 'timestamp': 1716994698939,
    'tenant': '1',
    'token': 'h622inzcXnQ1vzRaB58045Zk6GQ69UDP',
}
param=json.dumps(json_data.get("param"))
# param=json_data.get("param")
print(param)
get_params=execjs.compile(open('hongren.js','r',encoding='utf-8').read()).call("get_params",param)
json_data['timestamp']=get_params[0]
json_data['sign']=get_params[1]
print(json_data)

response = requests.post('https://ucp.hrdjyun.com:60359/api/dy', headers=headers, json=json_data)
print(response.text)